packages/contracts/ts/deploy.ts

```diff
@@ -607,20 +607,20 @@ export const deployPoseidonContracts = async (
 /**
  * Deploy a contract with linked libraries
  * @param contractFactory - the contract factory to use
- * @param name - the name of the contract
- * @param quiet - whether to suppress console output
+ * @param signer - the signer to use for deployment (optional)
  * @param args - the constructor arguments of the contract
  * @returns the deployed contract instance
  */
 export const deployContractWithLinkedLibraries = async <T extends BaseContract>(
   contractFactory: ContractFactory,
+  signer?: Signer,
```

Comment:
So I added the `signer` as a function parameter here. I also needed to send the `signer` in a couple of function calls (e.g. in line 623) but most of the `deployContractWithLinkedLibraries` calls are only sending the factories and therefore no type errors were generated there